lib/aws-manager.js

```diff
@@ -81,142 +82,164 @@ class AwsManager {
     let pricingStartDate = new Date();
     pricingStartDate.setMinutes(pricingStartDate.getMinutes() - 30);
 
-    let livingInstances;
-    let livingSpotRequests;
-    let deadInstances;
-    let deadSpotRequests = {};
-    let allPricingHistory;
-    let availableAZ;
-
-    try {
-      livingInstances = await this.ec2.describeInstances({
-        Filters: [
-          {
-            Name: 'key-name',
-            Values: [this.keyPrefix + '*'],
-          },
-          {
-            Name: 'instance-state-name',
-            Values: ['running', 'pending'],
-          },
-        ],
-      });
-      debug('fetched living instance state');
+    this.__previousApiState = this.__apiState;
+    let apiState = {
+      instances: [],
+      requests: [],
+    };
+    let deadState = {
+      instances: [],
+      requests: [],
+    };
 
-      livingSpotRequests = await this.ec2.describeSpotInstanceRequests({
-        Filters: [
-          {
-            Name: 'launch.key-name',
-            Values: [this.keyPrefix + '*'],
-          }, {
-            Name: 'state',
-            Values: ['open'],
-          },
-        ],
-      });
-      debug('fetched living spot request state');
+    let stalledKills = [];
+    let availableAZ = {};
+    let allPricingHistory = {};
+
+    // It would be better to do it by classification than by region for
+    // better concurrency, but this is easier and not too slow considering
+    // the 75s iteration frequency
+    _.forEach(this.ec2, async (obj, region) => {
```

Comment:
Here we have a high risk of dropping the ball, well dropping the promise :)

You'll want to do:
```js
await Promise.all(_.map(this.ec2, async (ec2, region) => {
   ....
}));
```

This way anything that goes wrong in here will propagate up... `_.forEach(, ...async ()` is always a footgun, a major one...
Technically, this could work fine, if you are confident that there are no bugs in the async function, but we don't have to be confident about that... So let's not even try...

nit: also using `ec2` instead of `obj` is much nicer :)